js/rank_states_utils.js

```diff
@@ -135,17 +136,36 @@ function rankEm() {
           .transition().duration(600)
             .attr('transform','translate(0,0)'); 
       } else {
-        barchoice
-          .on('mouseover', null)
-          .on('mouseout', null)
-          .attr('class','closed-rank-bar');
-        d3.select(barchoice.node().parentNode).select('text')
-          .classed('open-bar-name',false);
-        d3.select(this).style('opacity', 0)
-          .attr('transform','scale(0)'); 
+        
+        var thisBarname = d3.select(barchoice.node().parentNode).select('text').text();
```

Comment:
so that first select becomes the group that each rect and text exist in, right? The `d3.select(barchoice.node().parentNode)`? Just still learning all these ways of selecting things :)

Reply:
Yeah, so each rect is within a "parent" `g` that has the transform in it, and contains rect and text. I use this to select the text element in the same group as the rect that was selected. Not sure if there is a better way to do that. 
here, `parentNode` is the `g` group, and then I select the only text element that belongs to it.